IdentifierTests/KIFUITestActor-IdentifierTests.m

```diff
@@ -158,25 +159,28 @@ - (void)setValue:(float)value forSliderWithAccessibilityIdentifier:(NSString *)a
 - (void)waitForFirstResponderWithAccessibilityIdentifier:(NSString *)accessibilityIdentifier
 {
 	[self runBlock:^KIFTestStepResult(NSError **error) {
-		UIResponder *firstResponder = [[[UIApplication sharedApplication] keyWindow] firstResponder];
-		if ([firstResponder isKindOfClass:NSClassFromString(@"UISearchBarTextField")]) {
-			do {
-				firstResponder = [(UIView *)firstResponder superview];
-			} while (firstResponder && ![firstResponder isKindOfClass:[UISearchBar class]]);
-		}
-		UIResponder<UIAccessibilityIdentification>* firstResponderIdentification = nil;
-		if ([firstResponder conformsToProtocol:@protocol(UIAccessibilityIdentification)])
-		{
-			firstResponderIdentification = (UIResponder<UIAccessibilityIdentification>*)firstResponder;
-		}
-		else
-		{
-			[self failWithError:[NSError KIFErrorWithFormat:@"First responder does not conform to UIAccessibilityIdentification %@",  NSStringFromClass([firstResponder class])] stopTest:YES];
-			
-		}
-		KIFTestWaitCondition([[firstResponderIdentification accessibilityIdentifier] isEqualToString:accessibilityIdentifier],
+        BOOL didMatch = NO;
+        NSString *foundIdentifier = nil;
+        for (UIResponder __strong *firstResponder in [[UIApplication sharedApplication] firstResponders]) {
```

Comment:
Why. is the `__strong` needed here?

Reply:
It lets me reassign to `firstResponder` in the `for` loop without having to declare a temporary local variable.

Reply:
I'd probably still opt to use the local variable to make the logic more straightforward, but don't feel strongly on this.